contracts/non-fungible-tokens/andromeda-crowdfund/src/contract.rs

```diff
@@ -224,6 +231,61 @@ fn execute_remove_tier(ctx: ExecuteContext, level: Uint64) -> Result<Response, C
     Ok(resp)
 }
 
+fn execute_start_campaign(
+    ctx: ExecuteContext,
+    start_time: Option<MillisecondsExpiration>,
+    end_time: MillisecondsExpiration,
+    presale: Option<Vec<TierOrder>>,
+) -> Result<Response, ContractError> {
+    let ExecuteContext {
+        deps, info, env, ..
+    } = ctx;
+
+    // Only owner can start the campaign
+    let contract = ADOContract::default();
+    ensure!(
+        contract.is_contract_owner(deps.storage, info.sender.as_str())?,
+        ContractError::Unauthorized {}
+    );
+
+    // At least one tier should have no limit to start the campaign
+    ensure!(is_valid_tiers(deps.storage), ContractError::InvalidTiers {});
```

Comment:
We need to make sure that the current tiers are enough to hit the required funds for the crowdfund (if the field is set).

i.e. Sum of the amount of funds each tier can produce (number of tokens x token price)

Reply:
At least one tier has no limit as discussed, we can ensure that the campaign will hit the soft capital without problem